app/components/Texting/Texting.jsx

```diff
@@ -0,0 +1,223 @@
+import React, { useState } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import './Texting.scss';
+import { images } from 'assets';
+import * as dataService from '../../utils/DataService';
+import Loader from '../ui/Loader';
+
+// Initial state's values
+const initialState = {
+  userName: '',
+  phoneNumber: '',
+  agreed: false,
+  isLoading: false,
+  isSent: false,
+  error: false,
+};
+
+// Text resource informations to the user phone
+
+const Texting = ({ toggle, resource }) => {
+  const [state, setState] = useState(initialState);
```

Comment:
It's generally not recommended to set multiple properties like this in a single `useState()` variable. See the [docs](https://reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables) for some more detailed information, but you will usually want a separate `useState()` variable for each state variable. In some cases it makes sense to combine values into the same state variable, but usually only if are related and often change together.